volatility3/framework/plugins/linux/lsblk.py

```diff
@@ -0,0 +1,447 @@
+from typing import List
+import logging
+
+from volatility3.framework import interfaces, renderers, constants, exceptions
+from volatility3.framework.configuration import requirements
+from volatility3.framework.objects import utility
+
+vollog = logging.getLogger(__name__)
+
+
+class lsblk(interfaces.plugins.PluginInterface):
+    """Lists the block devices present in a particular linux memory image."""
+
+    _required_framework_version = (2, 0, 0)
+
+    _version = (2, 2, 1)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Linux kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+        ]
+
+    def _generator(
+        self,
+    ):
+        """Generates the list of block devices."""
+        vmlinux = self.context.modules[self.config["kernel"]]
+        try:
+            class_kset = vmlinux.object_from_symbol("class_kset")
+        except exceptions.SymbolError:
+            class_kset = None
+        if not class_kset:
+            raise TypeError(
+                "This plugin requires the class_kset structure. This structure is not present in the supplied symbol table. This means you are either analyzing an unsupported kernel version or that your symbol table is corrupt."
+            )
+
+        block_class = vmlinux.object_from_symbol("block_class")
+
+        subsys_off = vmlinux.get_type("subsys_private").relative_child_offset("subsys")
+        kobj_off = vmlinux.get_type("kset").relative_child_offset("kobj")
+
+        if not vmlinux.has_type("hd_struct"):
+            block_device_offset = vmlinux.get_type(
+                "block_device"
+            ).relative_child_offset("bd_device")
+        else:
+            hd_struct_offset = vmlinux.get_type("hd_struct").relative_child_offset(
+                "__dev"
```

Comment:
Is this fragile or does `__dev` exist in all versions of `hd_struct`?

Reply:
It is in all versions of hd_struct